packages/components/src/components/form/SegmentSlider/SegmentSlider.tsx

```diff
@@ -0,0 +1,242 @@
+import styled, { css } from 'styled-components';
+
+import { borders, spacingsPx, typography } from '@trezor/theme';
+
+const DEFAULT_SEGMENT = { max: 0, name: '' };
+
+const thumb = css<{ disabled?: boolean }>`
+    appearance: none;
+    background: white;
+    border-radius: ${borders.radii.full};
+    box-shadow: 0 0 4px 0 rgb(0 0 0 / 50%);
+    margin-top: calc((${spacingsPx.xxs} - ${spacingsPx.xl}) / 2);
+    width: ${spacingsPx.xl};
+    height: ${spacingsPx.xl};
+    cursor: ${({ disabled }) => !disabled && 'grab'};
+
+    ${({ disabled }) =>
+        !disabled &&
+        css`
+            &:active {
+                box-shadow: 0 0 2px 0 rgb(0 0 0 / 50%);
+                cursor: grabbing;
+            }
+        `}
+`;
+
+const StyledSegmentSlider = styled.div`
+    width: 100%;
+    position: relative;
+`;
+
+const SegmentInput = styled.input<{ disabled?: boolean }>`
+    margin: ${spacingsPx.sm} 0 ${spacingsPx.xs};
+    padding: 10px 0;
+    width: 100%;
+    vertical-align: top; /* prevent extra bottom space in Firefox */
+
+    position: relative;
+    z-index: 10;
+
+    width: 100%;
+
+    background: none;
+    appearance: none;
+    cursor: ${({ disabled }) => !disabled && 'pointer'};
+
+    &::-webkit-slider-runnable-track {
+        appearance: none;
+        background: transparent;
+    }
+
+    &::-webkit-slider-thumb {
+        ${thumb};
+    }
+
+    &::-moz-range-track {
+        appearance: none;
+        background: transparent;
+    }
+
+    &::-moz-range-thumb {
+        ${thumb};
+        transform: translateY(-6px); /* Firefox hack because of position absolute */
+    }
+
+    &:disabled {
+        pointer-events: auto;
+        cursor: not-allowed;
+    }
+`;
+
+const Segments = styled.div`
+    position: absolute;
+    top: 0;
+
+    margin: ${spacingsPx.sm} 0 ${spacingsPx.xs};
+    padding: 10px 0;
+    width: 100%;
+`;
+
+const Segment = styled.div<{ $start: number; $end: number }>`
```

Comment:
would be better to call it little bit differently so that it does not overlap with type `Segment`